coordinator/app/src/main/kotlin/linea/coordinator/config/v2/toml/CoordinatorConfigFilesToml.kt

```diff
@@ -22,10 +24,14 @@ data class CoordinatorConfigFileToml(
   val api: ApiConfigToml = ApiConfigToml(),
 )
 
-data class TracesLimitsConfigFileToml(
+data class TracesLimitsConfigFileV2Toml(
   val tracesLimits: Map<TracingModuleV2, UInt>,
 )
```

Comment:
Have you tried to use generics instead?

```suggestion
data class TracesLimitsConfigFileToml<T>(
   val tracesLimits: Map<T, UInt>,
)
```


```kotlin
data class CoordinatorConfigToml(
  val configs: CoordinatorConfigFileToml,
  val tracesLimitsV2: TracesLimitsConfigFileToml<TrancingModuleV2>?,
  val tracesLimitsV4: TracesLimitsConfigFileToml<TrancingModuleV4>?,
```

If it works, great otherwise don't invest much on it... I am ok the way it is